### PR TITLE
Improve navbar horizontal alignment FULLSCREEN (+ clear .gitignore)

### PR DESCRIPTION
**Pull request by dominklenda**
Hello,
I changed the position of navbar items in the full-screen mode. Basically, it widens the whole navbar and moves it a bit to the right when fullscreen mode is on. I guess, there are better solutions, e.g. align the left side of navbar with the left side of ArticleList. But this is the simplest I found at the moment.

Additionally, I removed files from .gitignore following [advice of @mickaobrien](https://github.com/ScienceCommons/curate_science/commit/b269476ad2d3ad3067864bd18d194c70040286d8#r38210189).

### CODE REVIEW

**Comment by dominklenda** [comment]
> perhaps better to consider using padding-left and padding-right, so the padding is always symmetrical horizontally? @mickaobrien any thoughts on this, and the updates in EmbeddedViewer.jsx?

In the [screenshot](https://user-images.githubusercontent.com/4512335/73439432-def07780-434f-11ea-9b60-b7b9faecaff7.png), the topbar looked asymmetrical, but, yes, symmetrical would be better, i guess. I changed it and added paddings. What do you think about this now?

**Comment by mickaobrien** [comment]
I think I'm missing what this is trying to solve - is there an issue that explains it? Is it related to the TopBar stuff in #95?

The TopBar should be centred horizontally by the CSS [`margin: 0 auto`](https://github.com/ScienceCommons/curate_science/blob/master/src/App.css#L37) in normal mode and fullscreen mode. I think it might look a little bit off because the padding of the hamburger menu on the left and the login button or user dropdown on the right don't match exactly.

**Comment by eplebel** [comment]
This is fixing the NAVBAR items not correctly being horizontally centered in full-screen mode, on all pages, see 1st item in #96:

improve horizontal alignment of NAVBAR items for all pages in FULL-SCREEN mode. That is, Recent page in FULL-SCREEN mode, search results page in full-screen mode, article page in full-screen mode, and author page in full-screen mode (so you have to GO INTO full-screen mode on all of those pages to see the alignment issue). See screenshot for horizontal alignments needed for author page in full-screen mode)

File: src/App.css [hunk 1]
```diff
@@ -56,6 +56,15 @@ textarea {
...
 
+.TopBar.EmbeddedViewerVisible {
+  width: 100%;
+  /*position: absolute;*/
+  padding-left: 100px;
+  padding-right: 100px;
+  /*position: relative;*/
+  /*left: 153px*/
+}
+
```
Comment (mickaobrien) [inline src/App.css:67]: I think you should remove the commented out lines here if you're happy with the new implementation. Things can get pretty hard to read if you leave all the old code there! :smile:

**Comment by mickaobrien** [comment]
Ah cool, thanks for that. So you want the logo + menu bit left-aligned and the help + user button right-aligned? If that's the case, setting the TopBar width to 100% (with some padding if necessary) as done should do the job! Looks good @dominklenda.

The one thing I'm not sure about is only doing it for full-screen mode. I think a consistent layout would be better, the way things jump when you go in and out of full-screen mode looks a bit off to me.

**Comment by eplebel** [comment]
> Ah cool, thanks for that. So you want the logo + menu bit left-aligned and the help + user button right-aligned? If that's the case, setting the TopBar width to 100% (with some padding if necessary) as done should do the job! Looks good @dominklenda.
> 
> The one thing I'm not sure about is only doing it for full-screen mode. I think a consistent layout would be better, the way things jump when you go in and out of full-screen mode looks a bit off to me.

good catch. indeed fix should be placed in the general TopBar class, thanks.

File: src/App.css (hunk 1 above)
Comment (dominklenda) [reply src/App.css:67]: Oh yes, sorry, that's a pretty mashed up :sweat_smile:

**Comment by dominklenda** [comment]
> Thanks for your input @mickaobrien! This is fixing the NAVBAR items not correctly being horizontally centered in full-screen mode, on all pages, see 1st item in #96:
> 
> improve horizontal alignment of NAVBAR items for all pages in FULL-SCREEN mode. That is, Recent page in FULL-SCREEN mode, search results page in full-screen mode, article page in full-screen mode, and author page in full-screen mode (so you have to GO INTO full-screen mode on all of those pages to see the alignment issue). See screenshot for horizontal alignments needed for author page in full-screen mode)

@eplebel thank you for clearing this up! Next time I'll attach the issue number and description of the fix to the pull request so others can navigate it easily.

**Comment by dominklenda** [comment]
> > Ah cool, thanks for that. So you want the logo + menu bit left-aligned and the help + user button right-aligned? If that's the case, setting the TopBar width to 100% (with some padding if necessary) as done should do the job! Looks good @dominklenda.
> > The one thing I'm not sure about is only doing it for full-screen mode. I think a consistent layout would be better, the way things jump when you go in and out of full-screen mode looks a bit off to me.
> 
> good catch. indeed fix should be placed in the general TopBar class, thanks.

@mickaobrien @eplebel Thanks, good points! Right, we need a consistent layout and the fix should be in the .TopBar class.

One thing is not clear to me:
**Do we want consistently symmetrical or asymmetrical ("fullscreen-centered") layout?**

Asymmetrical (`padding-left = 125px` and `padding-right = 15px`)
*Normal mode*
125_15n
*Fullscreen mode*
125_15f

**OR**

Symmetrical (`padding-left = 125px` and `padding-right = 125px`)
*Normal mode*
125_125n
*Fullscreen mode*
125_125f

I prefer consistently symmetrical.

**Comment by eplebel** [comment]
actually, upon second thought (after playing around with the different options), i think it *is* actually slightly better to update the .TopBar width to 100% *only* when embed viewer is visible,  OR else it looks pretty ugly when embed viewer is hidden.
(though i think there *IS* a better solution using horizontally-centered containers with max-widths, but this solution is good enough for now).

So please go ahead and revert back to:
```
.TopBar .EmbeddedViewerVisible {
  width: 100%;
  position: relative;
}
```
And without any padding.

Oh and somehow I just noticed that you're initial update (with left:63px, etc.) somehow may have sneaked into the production branch (see screenshot), perhaps via the bundle.js???
Anyway we should be more careful of that moving forward. (and now that you're using more proper branching approach, this might not be a problem moving forward)

image

**Comment by dominklenda** [comment]
> i think it is actually slightly better to update the .TopBar width to 100% only when embed viewer is visible

@eplebel yes, it looks much better!

> though i think there IS a better solution using horizontally-centered containers with max-widths, but this solution is good enough for now

ok, sure, we can discuss and apply it in the near future.

> Oh and somehow I just noticed that you're initial update (with left:63px, etc.) somehow may have sneaked into the production branch (see screenshot), perhaps via the bundle.js??? Anyway we should be more careful of that moving forward. (and now that you're using more proper branching approach, this might not be a problem moving forward)

oh yes, sorry, i was working on this one earlier and accidentally included it in the bundle.js. anyway, now with the branching git workflow everything should be controlled properly

File: src/App.css [hunk 2]
```diff
@@ -35,7 +35,7 @@ textarea {
 .TopBar {
   width: 1100px;
   margin: 0 auto;
-  position: relative;
+  positon: relative;
```
Comment (eplebel) [inline src/App.css:38]: please fix typo

**Comment by eplebel** [comment]
> Could you elaborate on this? Why this is a better solution?
for really highresolution monitors (UHD larger than 2000pixels wide),  100% widthwill start looking ugly.  So you'd need a container with max-width of 2000px ish, but this is rare and so too low priority to tackle this right now!

**Comment by dominklenda** [comment]
> why the updates to the package-lock.json?

after updating my OS, I had to update npm packages related to the app (i just `npm install`ed them). This produced updated package-lock.json, which I previously ignored. But following [npm's official docs ](https://docs.npmjs.com/files/package-lock.json), [stack users](https://stackoverflow.com/questions/44206782/do-i-commit-the-package-lock-json-file-created-by-npm-5), and @mickaobrien's [comment](https://github.com/ScienceCommons/curate_science/commit/b269476ad2d3ad3067864bd18d194c70040286d8#r38210189) I committed it. It improves the readability of the changes in the node modules. 
But if you wish I can revert a commit and update only Bundle.js. It looks like keeping package-lock.json updated is not that important.

**Comment by eplebel** [comment]
> > why the updates to the package-lock.json?
> 
> after updating my OS, I had to update npm packages related to the app (i just `npm install`ed them). This produced updated package-lock.json, which I previously ignored. But following [npm's official docs ](https://docs.npmjs.com/files/package-lock.json), [stack users](https://stackoverflow.com/questions/44206782/do-i-commit-the-package-lock-json-file-created-by-npm-5), and @mickaobrien's [comment](https://github.com/ScienceCommons/curate_science/commit/b269476ad2d3ad3067864bd18d194c70040286d8#r38210189) I committed it. It improves the readability of the changes in the node modules.
> But if you wish I can revert a commit and update only Bundle.js. It looks like keeping package-lock.json updated is not that important.

ok, no that's fine, just wanted to know why, thanks.

**Pull request merged**